WordPress/gutenberg
# Show UI warning to users with insufficient permissions when trying to create a new Navigation

## Description
Explores part of https://github.com/WordPress/gutenberg/issues/36286#issuecomment-990845661 to display a warning to the user if they have insufficient permissions to _**publish**_ a given Navigation block.

Publish status is not then ability to create a post. Rather it is the ability to publish it. By default only admin Role users can do this.

https://github.com/WordPress/wordpress-develop/blob/424afb85afa2e19f30d9836440f63c5237b4102e/src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php#L2115-L2116

Moreover the Publish capability is a Meta capability meaning it is on a per post basis and not a global thing (i.e. you need a Post ID to check against for "publish" capability).

As things stand `canUser` cannot determine publish status so this PR updates that to enable the abiltity to do so.

Moreover, when we're creating the initial Navigation post, because we're attempting to do so with the `publish` status, the POST request fails and thus we never have a Post ID to check against for "publish" permission. This PR amends the initial creation to only create a _draft_ Navigation Post. This allows us to get a Post ID back and use that for the `canUser` publish check. We'd have to amend the block to have a step which changes the post status to `publish` once the initial creation is complete. 

Note this doesn't handle editing existing Navigations. That's handled in https://github.com/WordPress/gutenberg/pull/37286.

## How has this been tested?




- Install a Plugin such as User Switching to afford easy switching between Users.
- Create a Admin and Contributor level user.
- As the Contributor, create a Post.
- As the Contributor go to that post and add a Navigation block.
- Click "Start empty" in the placeholder.
- You should see a warning saying you cannot save the Nav block.

## Screenshots 

https://user-images.githubusercontent.com/444434/145603875-961813bc-2233-4d1b-8f3f-f6b80d3f780f.mp4


## Types of changes





## Checklist:
